core/src/genesis.rs

```diff
@@ -37,13 +35,56 @@ pub fn genesis() -> core::Block {
 				tm_mday: 4,
 				..time::empty_tm()
 			},
-			difficulty: Difficulty::from_num(MINIMUM_DIFFICULTY),
-			total_difficulty: Difficulty::from_num(MINIMUM_DIFFICULTY),
-			utxo_root: empty_hash,
-			range_proof_root: empty_hash,
-			kernel_root: empty_hash,
 			nonce: global::get_genesis_nonce(),
-			pow: core::Proof::zero(proof_size), // TODO get actual PoW solution
+			..Default::default()
+		},
+		inputs: vec![],
+		outputs: vec![],
+		kernels: vec![],
+	}
+}
+
+/// First testnet genesis block, still subject to change (especially the date,
+/// will hopefully come before Christmas).
+pub fn genesis_testnet1() -> core::Block {
+	core::Block {
+		header: core::BlockHeader {
+			height: 0,
+			previous: core::hash::Hash([0xff; 32]),
+			timestamp: time::Tm {
+				tm_year: 2017 - 1900,
+				tm_mon: 12,
+				tm_mday: 25,
+				..time::empty_tm()
+			},
+			nonce: 1,
+			pow: core::Proof::zero(consensus::PROOFSIZE),
+			..Default::default()
+		},
+		inputs: vec![],
+		outputs: vec![],
+		kernels: vec![],
+	}
+}
+
+/// Placeholder for mainnet genesis block, will definitely change before
+/// release so no use trying to pre-mine it.
+pub fn genesis_main() -> core::Block {
```

Comment:
Maybe just be explicit about each genesis block? `main|testnet1|user_testing|automated_testing`?

Reply:
Seemed that everything non-public could share the same, as they'd have the exact same definition.